### PR TITLE
Elevation in bars

### PR DESCRIPTION
**Pull request by rajkothari634**
Shadow and elevation is completed

### CODE REVIEW

**Comment by ShivangiSingh17** [comment]
@rajkothari634 Please go through the contributing guidelines and give proper description while creating the PR. Also please solve the conflicts and squash your commits.

**Comment by rajkothari634** [comment]
Okay I will do it tonight

File: library/src/main/java/in/krharsh17/barview/ViewUtils.java [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+
+import androidx.annotation.ColorRes;
+import androidx.annotation.DimenRes;
+import androidx.core.content.ContextCompat;
+
+import static android.view.View.LAYER_TYPE_SOFTWARE;
+
+/**
+ * Created by ArmanSo on 4/16/17.
+ */
```
Comment (krharsh17) [inline library/src/main/java/in/krharsh17/barview/ViewUtils.java]: Please remove this

File: library/src/main/java/in/krharsh17/barview/BarGroup.java [hunk 2]
```diff
@@ -38,7 +48,37 @@
...
+            "#22",
+            "#24",
+            "#26",
+            "#28",
+            "#30",
+            "#32",
+            "#35",
+            "#"
+    };
+    //ltrb
```
Comment (krharsh17) [inline library/src/main/java/in/krharsh17/barview/BarGroup.java]: What does this mean?

File: library/src/main/java/in/krharsh17/barview/BarGroup.java [hunk 3]
```diff
@@ -38,7 +48,37 @@
...
+            "#20",
+            "#22",
+            "#24",
+            "#26",
+            "#28",
+            "#30",
+            "#32",
+            "#35",
+            "#"
+    };
```
Comment (krharsh17) [inline library/src/main/java/in/krharsh17/barview/BarGroup.java:83]: Please provide documentation for this

File: library/src/main/java/in/krharsh17/barview/BarGroup.java [hunk 4]
```diff
@@ -128,28 +168,148 @@ void setupLabel() {
     }
 
     void setupInitial() {
-        initialParams = new LayoutParams(dp(12), dp(BAR_HEIGHT));
+
+        int j = 0;
```
Comment (krharsh17) [inline library/src/main/java/in/krharsh17/barview/BarGroup.java]: Better naming conventions are expected

File: library/src/main/java/in/krharsh17/barview/Bar.java [hunk 5]
```diff
@@ -6,23 +6,39 @@
...
         ViewGroup.LayoutParams params = this.getLayoutParams();
-        params.width = Math.round(params.width * (progress));
+        params.width   = Math.round(params.width * (progress)) + increaseWidth;
         this.setLayoutParams(params);
         invalidate();
         requestLayout();
     }
+    public Bar(Context context) {
+        super(context);
+//initBackground();
```
Comment (krharsh17) [inline library/src/main/java/in/krharsh17/barview/Bar.java]: Format entire code properly

**Review by krharsh17 — COMMENTED** [review]
Please provide documentation with all the new methods/variables that you have introduced. Also, format your code properly and squash all your commits into one. Link the issue that you have addressed, provide a sample of the change that you've made, describing how to use it from XML and resolve all merge conflicts.

File: library/src/main/java/in/krharsh17/barview/BarGroup.java (hunk 3 above)
Comment (rajkothari634) [reply library/src/main/java/in/krharsh17/barview/BarGroup.java:83]: documentation done

File: library/src/main/java/in/krharsh17/barview/BarGroup.java (hunk 4 above)
Comment (rajkothari634) [reply library/src/main/java/in/krharsh17/barview/BarGroup.java]: naming conventions are improved

File: library/src/main/java/in/krharsh17/barview/Bar.java (hunk 5 above)
Comment (rajkothari634) [reply library/src/main/java/in/krharsh17/barview/Bar.java]: unused comments are erased and formatting is done

File: library/src/main/java/in/krharsh17/barview/BarGroup.java (hunk 2 above)
Comment (rajkothari634) [reply library/src/main/java/in/krharsh17/barview/BarGroup.java]: comments are added to explain it

File: library/src/main/java/in/krharsh17/barview/ViewUtils.java (hunk 1 above)
Comment (rajkothari634) [reply library/src/main/java/in/krharsh17/barview/ViewUtils.java]: View Utils file is not needed so we can delete this

**Pull request closed without merging**